### PR TITLE
[YUNIKORN-1499] Use fine-grained K8s access control in example deployments.

### PR DESCRIPTION
**Pull request by zhuqi-lucas**
### What is this PR for?
When deploying yunikorn-admission-controller with 
 yunikorn-admission-controller-rbac, it shows can't list and watch for PriorityClass in pod log.   
 
 Besides this, we should use fine-grained K8s access control in example deployments.

### What type of PR is it?
* [x] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/YUNIKORN-1499
* Put link here, and add [YUNIKORN-*Jira number*] in PR title, eg. `[YUNIKORN-2] Gang scheduling interface parameters`

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Review by craigcondit — CHANGES_REQUESTED** [review]
Looks fine, but can we fix the scheduler rbac.yaml as well (both the scheduler and admission controller read the PriorityClass objects).

**Comment by zhuqi-lucas** [comment]
Thank you, @craigcondit, for review, in the scheduler rbac.yaml , we already used the cluster-admin cluster role which will cover the PriorityClass access.
 
`apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: yunikorn-rbac
subjects:
  - kind: ServiceAccount
    name: yunikorn-admin
roleRef:
  kind: ClusterRole
  name: cluster-admin
  apiGroup: rbac.authorization.k8s.io
`

**Comment by wilfred-s** [comment]
We should not be using the cluster admin role anywhere. That was dropped as part of the [YUNIKORN-997](https://issues.apache.org/jira/browse/YUNIKORN-997) in release 1.0. Looks like we forgot to merge those changes intothe non helm examples.

**Comment by zhuqi-lucas** [comment]
> We should not be using the cluster admin role anywhere. That was dropped as part of the [YUNIKORN-997](https://issues.apache.org/jira/browse/YUNIKORN-997) in release 1.0. Looks like we forgot to merge those changes intothe non helm examples.

Thank you @wilfred-s for this reminder, i will cover this in this PR, and i will also create a new PR to check the helm chart priorityclass related things.

**Comment by wilfred-s** [comment]
The helm charts have been updated via: apache/yunikorn-release#120
A copy of the charts is used during the e2e tests so without that change none of the e2e tests would work.

**Comment by zhuqi-lucas** [comment]
It makes sense to me @wilfred-s  , so i will cover the fine-grained and priority in example deployments, thanks!

File: deployments/scheduler/yunikorn-rbac.yaml [hunk 1]
```diff
@@ -26,7 +66,49 @@ metadata:
...
+kind: RoleBinding
+metadata:
+  name: yunikorn-rbac
+subjects:
+  - kind: ServiceAccount
+    name: yunikorn-admin
+roleRef:
+  kind: Role
+  name: yunikorn-scheduler
+  apiGroup: rbac.authorization.k8s.io
```
Comment (wilfred-s) [inline deployments/scheduler/yunikorn-rbac.yaml:114]: please make sure that we we have. a newline at the end of the file.

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: deployments/scheduler/yunikorn-rbac.yaml (hunk 1 above)
Comment (zhuqi-lucas) [reply deployments/scheduler/yunikorn-rbac.yaml]: Sure, add a newline now.

**Review by wilfred-s — APPROVED** [review]
LGTM

**Pull request closed without merging**